build_scripts/build_zlib.sh

```diff
@@ -16,11 +16,21 @@
 
 #zlib
 pushd third_party/zlib
+mkdir -p build
+cd build
+echo "set(CMAKE_SYSTEM_NAME Linux)" > toolchain.cmake
+echo "set(CMAKE_SYSTEM_PROCESSOR ${CMAKE_TARGET_ARCH})" >> toolchain.cmake
+echo "set(CMAKE_C_COMPILER ${CC_COMP})" >> toolchain.cmake
+    CFLAGS="-fPIC" \
+    CXXFLAGS="-fPIC" \
+    CC=${CC_COMP} \
+    CXX=${CXX_COMP} \
+cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE=toolchain.cmake \
+          -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX} ..
     CFLAGS="-fPIC" \
     CXXFLAGS="-fPIC" \
     CC=${CC_COMP} \
     CXX=${CXX_COMP} \
-./configure --prefix=${INSTALL_PREFIX}
```

Comment:
Configure yields wrong compilation command when CC and CXX is provided in the most recent version (it doesn't set SONAME in the library). cmake seems to work fine.